include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1620,6 +1622,82 @@ namespace internal
 } // namespace internal
 
 
+
+namespace MGTransferGlobalCoarseningTools
+{
+  template <int dim, int spacedim>
+  std::vector<std::shared_ptr<Triangulation<dim, spacedim>>>
+  create_geometric_coarsening_sequence(
+    const Triangulation<dim, spacedim> &fine_triangulation_in)
+  {
+    std::vector<std::shared_ptr<Triangulation<dim, spacedim>>>
+      coarse_grid_triangulations;
+
+#ifndef DEAL_II_WITH_P4EST
+    (void)fine_triangulation_in;
+    Assert(false, ExcNotImplemented());
+#else
+    const auto fine_triangulation =
+      dynamic_cast<const parallel::distributed::Triangulation<dim, spacedim> *>(
+        &fine_triangulation_in);
+
+    Assert(fine_triangulation, ExcNotImplemented());
+
+    const unsigned int max_level = fine_triangulation->n_global_levels() - 1;
+
+    const auto coarse_mesh_description =
+      GridTools::get_coarse_mesh_description(*fine_triangulation);
+
+    coarse_grid_triangulations.resize(fine_triangulation->n_global_levels());
+
+    coarse_grid_triangulations[max_level].reset(
+      const_cast<Triangulation<dim, spacedim> *>(&fine_triangulation_in),
+      [](auto &) {});
+
+    // create coarse meshes
+    for (unsigned int l = max_level - 1; l != static_cast<unsigned int>(-1);
+         --l)
```

Comment:
For what it's worth, elsewhere in the library we often use the loops constructed as
```suggestion
    for (unsigned int l = max_level; l > 0; )
      {
        --l;
```
Not sure if you have strong objections against this one; I prefer it because not all progammers are familiar with the wrap-around behavior of unsigned int specified in the C++ standard.